datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -223,23 +223,48 @@ class except the :py:meth:`check` method but sometimes it might be useful for a
         }
 
         # Setup metric limits
-        try:
-            metric_limit = int(self.instance.get('max_returned_metrics', self.DEFAULT_METRIC_LIMIT))
-            # Do not allow to disable limiting if the class has set a non-zero default value
-            if metric_limit == 0 and self.DEFAULT_METRIC_LIMIT > 0:
-                metric_limit = self.DEFAULT_METRIC_LIMIT
-                self.warning(
-                    'Setting max_returned_metrics to zero is not allowed, reverting to the default of %s metrics',
-                    self.DEFAULT_METRIC_LIMIT,
-                )
-        except Exception:
-            metric_limit = self.DEFAULT_METRIC_LIMIT
-        if metric_limit > 0:
-            self.metric_limiter = Limiter(self.name, 'metrics', metric_limit, self.warning)
+        self.metric_limiter = self._get_metric_limiter(self.name, instance=self.instance)
 
         # Functions that will be called exactly once (if successful) before the first check run
         self.check_initializations = deque([self.send_config_metadata])
 
+    def _get_metric_limiter(self, name, instance=None):
+        limit = self._get_metric_limit(instance=instance)
+
+        if limit > 0:
+            return Limiter(name, 'metrics', limit, self.warning)
+
+        return None
+
+    def _get_metric_limit(self, instance=None):
+        if instance is None:
+            # Agent 6+ will now always pass an instance when calling into a check, but we still need to
+            # account for this case due to some tests not always passing an instance on init.
+            return self.DEFAULT_METRIC_LIMIT
```

Comment:
Could you add a debug log here? Similar to this one

```
            self.warning(
                'Setting max_returned_metrics to zero is not allowed, reverting to the default of %s metrics',
                self.DEFAULT_METRIC_LIMIT,
            )
```

Reply:
Added!